### ISSUE TITLE
How to Examine Network Trafffic Generated by DNF5?

### ISSUE THREAD

**Issue by nobozo** [rpm-software-management/dnf5#1638]
I'm trying to do some experiments with DNF5. I've installed it on Fedora 40 (yes, I do mean DNF5).

The first thing I want to do is examine the network traffic sent by DNF5 so I setup mitmproxy and modified /etc/dnf/dnf.conf to use it as a proxy. I'm using https://localhost:8080 as the endpoint. I run something simple like "dnf5 update" and I don't see any network traffic but the command completes very quickly. Does DNF5 do some kind of caching so it doesn't always generate network traffic?

How would you suggest running DNF5 in a way that results in network traffic being generated?

(I know mitmproxy is working because I modified my Firefox config to use it, and I see what I expect.)

Thanks.

**Comment by m-blaha** [comment]
If the repository metadata are fresh, then there might be no need for any download. You can try `dnf5 update --refresh` to enforce cached metadata synchronization.

**Comment by nobozo** [comment]
Thanks. This is what I was looking for.

Then I ran into another problem, which was caused by dnf5 not trusting the certificate used by mitmproxy. When running dnf5 I would see

error checking if metalink "https://mirrors.fedoraproject.org/metalink?repo=updates-released-f40&arch=x86_64" is in sync for repository "updates"
 Librepo error: Cannot prepare internal mirrorlist: Curl error (60): SSL peer certificate or SSH remote key was not OK for https://mirrors.fedoraproject.org/metalink?repo=updates-released-f40&arch=x86_64 [SSL certificate problem: unable to get local issuer certificate]

This happened even though I followed the directions for installing the cert given in
[](https://docs.fedoraproject.org/en-US/quick-docs/using-shared-system-certificates/#proc_adding-new-certificates)

I tore out what little hair I have left for a while trying to resolve this. I eventually found [](https://github.com/mitmproxy/mitmproxy/discussions/5307#discussioncomment-9368794) which says

"Also note that when mitmproxy is run as root (for using privileged ports as 443/80), the initially, by default automatically generated CA certificates are not those in the normal user folder (when mitproxy was previously ran as normal user), but in root home directory (/root/), and those are randomly generated, so for root those differ to normal user."

I had been running mitmproxy as root so I tried running it as a normal user. This worked!!

I hope this saves somebody else from fighting this problem.
